alphagov/forms-runner
# Update cookies page and add cookie consent form

### What problem does this pull request solve?

Trello card: https://trello.com/c/1e6Mzt5m/1586-update-cookies-page-for-ga4-for-the-runner


Makes updates to the cookies page to reflect the analytics changes - largely copied from forms-product-pages:
- updates the content to include information about analytics cookies
- adds a cookie consent form so that users can set their preferences from this page
- removes the cookie banner from the page (since the consent form covers the same functionality)

Includes some changes similar to those in the [cookie banner PR](https://github.com/alphagov/forms-runner/pull/796):

- putting the analytics content on this page behind the `analytics_enabled` feature flag
- moving some of the logic out of application.js and into the cookie consent form component
- putting the cookie consent form component JS in a ViewComponent with the cookie banner HTML, instead of in the frontend folder

### Screenshots
Before:
![Cookies page, with a cookie banner at the top and a section for essential cookies. The cookie banner is the only way to set the preference for analytics cookies on this page.](https://github.com/alphagov/forms-runner/assets/5861235/50933759-8e0e-441a-ada4-df7f7264a8ff)


After:
![Cookies page, with no cookie banner at the top. The essential cookies section now includes an entry for analytics_consent, and there is a new section for analytics cookies. Under the analytics cookies section is a form for setting analytics cookies](https://github.com/user-attachments/assets/9cbe989e-a7e6-4d8c-b6cf-26b763bbed10)


### Testing instructions
You can see whether GA is being loaded by checking the `<head>` element to see if it contains the google tag manager script (`<script async="" src="https://www.googletagmanager.com/gtm.js?id=GTM-T2SJXKKQ"></script>`). You should only see this if you've accepted analytics cookies.

### Things to consider when reviewing



- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?
